verifiers/internal/gcb/verifier.go

```diff
@@ -40,9 +41,70 @@ func (v *GCBVerifier) VerifyArtifact(ctx context.Context,
 
 // VerifyImage verifies provenance for an OCI image.
 func (v *GCBVerifier) VerifyImage(ctx context.Context,
-	artifactImage string,
+	provenance []byte, artifactImage string,
 	provenanceOpts *options.ProvenanceOpts,
 	builderOpts *options.BuilderOpts,
 ) ([]byte, string, error) {
-	return nil, "todo", serrors.ErrorNotSupported
+	prov, err := ProvenanceFromBytes(provenance)
+	if err != nil {
+		return nil, "", err
+	}
+
+	// Verify signature on the intoto attestation.
+	if err = prov.VerifySignature(); err != nil {
+		return nil, "", err
+	}
+
+	// Verify intoto header.
+	if err = prov.VerifyIntotoHeaders(); err != nil {
+		return nil, "", err
+	}
+
+	// Verify the builder.
+	builderID, err := prov.VerifyBuilderID(builderOpts)
+	if err != nil {
+		return nil, "", err
+	}
+
+	// Verify subject digest.
+	if err = prov.VerifySubjectDigest(provenanceOpts.ExpectedDigest); err != nil {
+		return nil, "", err
+	}
+
+	// Verify source.
+	if err = prov.VerifySourceURI(provenanceOpts.ExpectedSourceURI); err != nil {
+		return nil, "", err
+	}
+
+	// Verify branch.
+	if provenanceOpts.ExpectedBranch != nil {
+		if err = prov.VerifyBranch(*provenanceOpts.ExpectedBranch); err != nil {
+			return nil, "", err
+		}
+	}
+
+	// Verify the tag.
+	if provenanceOpts.ExpectedTag != nil {
+		if err := prov.VerifyTag(*provenanceOpts.ExpectedTag); err != nil {
```

Comment:
there's likely no way to do this right? the only way would be an online tag comparison to the commit in the URI i think

Reply:
You're right. This will just report an error "unsupported" if the user passes this option. This way it's not silently ignored.
Wdut?

Reply:
Actually there is a way to do that: GCB provenance contains the tag in the subject. I have not implemented it yet.

Reply:
I take back what I say. There is confusion between repository tag and image tag. Container tag is do-able today, but I think it should just be a `-subject` option. Maybe we should rename the option to `-source-tag`? So we'd have `-source-uri`, `-source-tag`, and `-container-tag`...?

Wdut?

Reply:
Oh, I meant `tag` from source URI, like this was saying. Not container tag. But agree that we should split the option names. This is mentioned in the issue https://github.com/slsa-framework/slsa-verifier/issues/180

Since I'm breaking the API in the next PR we can do it in the next release.

But back on the `tag`: I thought the GCB provenance didn't include source tags, just the commit. At least it seemed so in
```
	if !strings.HasPrefix(uri, expectedSourceURI+"/commit/") {
		return fmt.Errorf("%w: expected '%s', got '%s'",
			serrors.ErrorMismatchSource, expectedSourceURI, uri)
	}
```

Reply:
You're correct. The `VerifyTag` function returns "not supported", like `VerifyBranch` and `VerifyVersionedTag`

Reply:
Yeah, that's a pretty hard trade-off. I liked the per-verifier README.md you suggested in the documentation PR, so I think it's best to just put there. Otherwise we need to fetch the verifier before we parse args.

Reply:
So I'll leave this as-in. Confirmed?